go/consensus/cometbft/full/archive.go

```diff
@@ -70,6 +72,12 @@ func (srv *archiveService) Start() error {
 		}
 	}()
 
+	// Start command dispatchers for all the service clients.
+	srv.serviceClientsWg.Add(len(srv.serviceClients))
+	for _, svc := range srv.serviceClients {
+		go srv.serviceClientWorker(srv.ctx, svc)
```

Comment:
```suggestion
		go func ()  {
			defer srv.serviceClientsWg.Done()
			srv.serviceClientWorker(srv.ctx, svc)
		}()
```
This way, additions and subtractions to the `WaitGroup` are in the same place (less prone to bugs), and the method `serviceClientWorker` is not dependent on it (can be tested without it).

But if you want to keep it in line with `fullService`, then just leave it as it is.

Reply:
Updated